urwid/display/_raw_display_base.py

```diff
@@ -639,34 +647,32 @@ def using_standout_or_underline(a: AttrSpec | str) -> bool:
                 elif y == maxrow - 1 and maxcol > 1:
                     row, back, ins = self._last_row(row)  # noqa: PLW2901
 
-            first = True
-            lasta = lastcs = None
             for a, cs, run in row:
                 if not isinstance(run, bytes):  # canvases render with bytes
                     raise TypeError(run)
 
                 if cs != "U":
                     run = run.translate(UNPRINTABLE_TRANS_TABLE)  # noqa: PLW2901
 
-                if first or lasta != a:
-                    o.append(attr_to_escape(a))
-                    lasta = a
+                if last_attributes != a:
+                    output.append(attr_to_escape(a))
+                    last_attributes = a
 
-                if not (IS_WINDOWS or IS_WSL) and (first or lastcs != cs):
+                if encoding != "utf-8" and (first or last_charset_flag != cs):
```

Comment:
In case of WSL and Windows, charger is always `utf-8`. Windows is mentioned in the python `stdin`/`stdout`/`stderr` documentation explicit, WSL is well known.
On the listed in the PR comment terminals this rules worked perfectly with non-ascii symbols to display.

Reply:
My experience is mostly in VT100/xterm type terminals rather than the Windows command.com environment, but I'll take a look.

Reply:
Windows part is good, "normal" modern terminals also good. But I'm not 100% sure about old terminals.
Reduced amount of escape sequences seriously reduces future work pain, but I do not want to break something.
In case of this PR fully OK, we can detach final encoding for render from `Canvas` (final composition & screen-independent render).